File: 백준/Silver/11057. 오르막 수/오르막 수.cc
#include<iostream>

using namespace std;

int N;
long long dp[1001][10] = { 0, };
long long sol = 0;


int main() {

	

	cin >> N;

	for (int i = 0; i < 10; i++) {
		dp[1][i] = 1;
	}

	for (int i = 2; i <= N; i++) {
		for (int j = 0; j <= 9; j++) {
			for (int t = 0; t <= j; t++) {
				dp[i][j] += dp[i - 1][t];
			}
			dp[i][j] %= 10007;
		}
	}

	for (int i = 0; i < 10; i++) {
		sol = sol + dp[N][i];
	}

	//cout << sol << '\n';
	cout << sol%10007;


	return 0;
}

File: 백준/Silver/11057. 오르막 수/오르막 수.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        int n = Integer.parseInt(br.readLine());
        int[][] dp = new int[n + 1][10];
        int sol=0;

        for (int i = 0; i <= 9; i++) {
            dp[1][i] = 1;
        }
        for (int i = 1; i < n; i++) {
            for (int j = 0; j <= 9; j++) {
                for (int k = 0; k <= j; k++) {
                    dp[i+1][j] += dp[i][k];
                }
                dp[i+1][j] %= 10007;
            }
        }

        for (int i = 0; i <= 9; i++) {
            sol += dp[n][i];
        }

        sol %= 10007;

        bw.write(sol + "");
        bw.flush();

        bw.close();

    }
}